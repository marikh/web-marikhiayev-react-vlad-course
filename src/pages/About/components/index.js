import AboutHeroArea from './AboutHeroArea';

export {
    AboutHeroArea
}
