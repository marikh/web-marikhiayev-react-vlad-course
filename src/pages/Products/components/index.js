import ProductsHeroArea from './ProductsHeroArea';

export {
    ProductsHeroArea
}
