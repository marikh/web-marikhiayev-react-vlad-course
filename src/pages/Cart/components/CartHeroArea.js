import React from 'react';
import { HeroArea, Heading } from '../../../components';

export default () => (
    <HeroArea>
        <Heading size={3}>
            Cart
        </Heading>
    </HeroArea>
) 