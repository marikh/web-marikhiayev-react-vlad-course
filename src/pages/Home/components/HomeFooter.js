import React from 'react';
import { SocialLinks } from '../../../components/';
import { withRouter } from 'react-router-dom';

export default withRouter(({ links, history, location, match }) => {
    return (
        <div>
            <SocialLinks links={links} />
            <h3 onClick={() => history.goBack()}>Back</h3>
        </div>
    )
})

