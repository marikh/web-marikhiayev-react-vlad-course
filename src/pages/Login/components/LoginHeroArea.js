import React from 'react';
import { HeroArea, Heading } from '../../../components';

export default () => (
    <HeroArea>
        <Heading size={3}>
            Login
        </Heading>
    </HeroArea>
) 