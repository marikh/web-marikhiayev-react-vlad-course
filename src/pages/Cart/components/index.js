import CartHeroArea from './CartHeroArea';

export {
    CartHeroArea
}
