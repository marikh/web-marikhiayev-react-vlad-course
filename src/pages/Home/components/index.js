import HomeFooter from './HomeFooter';
import HomeHeroArea from './HomeHeroArea';

export {
    HomeFooter,
    HomeHeroArea    
}
