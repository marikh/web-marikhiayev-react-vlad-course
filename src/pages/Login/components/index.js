import LoginHeroArea from './LoginHeroArea';

export {
    LoginHeroArea
}
