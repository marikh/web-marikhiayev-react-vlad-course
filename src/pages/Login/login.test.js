import { mount, shallow } from "enzyme";
import React from 'react';
import { MemoryRouter, Prompt } from 'react-router-dom'
import { Layout, Section } from '../../components/';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import ConnectedLogin,{Login} from '../../pages/Login/';
import {createStore} from 'redux';
import thunk from 'redux-thunk';
import { delay } from '../../common/Extensions/promises'

const MENU_INITIAL_STATE = {showProtectedLinks : false};
const LOGIN_INITIAL_STATE = {
  userName : "Marik",
  password : "M",
  loggingIn : false,
  loggedIn : false,
  loginFailed : false,
};


const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);

const initialState = { loginPage: LOGIN_INITIAL_STATE, menu: MENU_INITIAL_STATE } ;
    let store, wrapper;  

describe('login outer tests', () => {
    

     beforeEach(()=>{
        if(wrapper != null)
            wrapper.unmount();

        store = mockStore(initialState);
        wrapper = mount( <Provider store={store}><MemoryRouter><ConnectedLogin /></MemoryRouter></Provider> );
    })

    it("has Layout", () => {
        expect(wrapper.find(Layout).length).toBeGreaterThan(0)
    })
    
    it("has Section Component", () => {
        expect(wrapper.find(Section).length).toBeGreaterThan(0)
    })

    it(`check login click`, () => {

        const userLoggingInAction = { type: 'USER_LOGGING_IN' };
        const userLoggedInAction = { type: 'USER_LOGGED_IN', loggedInUser: "Marik" };

        const firstProductDeleteButton = wrapper.find('button[type="submit"]').first();
        firstProductDeleteButton.simulate('submit');

        // Test if my store dispatched the expected actions        
        const timeThresholdToSuccess  = 4000;
        const actions = store.getActions();
        
        return delay(timeThresholdToSuccess).then(() =>
            expect(actions).toEqual([userLoggingInAction, userLoggedInAction])
        );
    })

   it(`check login should fail`, () => {

        const failureInitialState = { loginPage: {
            userName : "M",
            password : "M",
            loggingIn : false,
            loggedIn : false,
            loginFailed : false,
        }, menu: MENU_INITIAL_STATE 
        } ;
        
        store = mockStore(failureInitialState);
        wrapper = mount( <Provider store={store}><MemoryRouter><ConnectedLogin /></MemoryRouter></Provider> );
        
        const userLoggingInAction = { type: 'USER_LOGGING_IN' };
        const loginFailedAction = { type: 'LOGIN_FAILED'};

        const firstProductDeleteButton = wrapper.find('button[type="submit"]').first();
        firstProductDeleteButton.simulate('submit');

        // Test if my store dispatched the expected actions
        const timeThresholdToSuccess  = 4000;
        const actions = store.getActions();
        

        return delay(timeThresholdToSuccess).then(() =>
            expect(actions).toEqual([userLoggingInAction, loginFailedAction])
        );
    })
})

describe('login Shallow Render',()=>{
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<Login />);
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1);
    });

    it('contains userName InputField', () => {
        expect(wrapper.find('InputField[name="userName"]').length).toEqual(1);
    });

    it('contains password InputField', () => {
        expect(wrapper.find('InputField[name="password"]').length).toEqual(1);
    });
})

describe('login snapshot tests', () => {

    it('test weird snapshot', () => {

        const store = mockStore(initialState);

        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <ConnectedLogin />
                </MemoryRouter>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
 })

