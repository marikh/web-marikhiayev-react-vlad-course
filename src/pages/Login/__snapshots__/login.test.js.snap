// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart snapshot tests test weird snapshot 1`] = `
<div
  className="App"
  onClick={[Function]}
>
  <header
    className="App-header"
  >
    <div
      className="App-menubar"
    >
      <label
        className="hamburger-icon fa fa-bars"
        onClick={[Function]}
      />
      <div
        className="menu-content"
      >
        <div
          className="links-container"
        >
          <a
            className="page-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Home
            </span>
          </a>
          <a
            className="page-link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
          <a
            className="page-link"
            href="/products"
            onClick={[Function]}
          >
            <span>
              Products
            </span>
          </a>
          <a
            className="page-link"
            href="/contact"
            onClick={[Function]}
          >
            <span>
              Contact
            </span>
          </a>
          <a
            className="page-link"
            href="/login"
            onClick={[Function]}
          >
            <span>
              SignIn
            </span>
          </a>
        </div>
        <div
          className="socialLinks"
        >
          <div
            className="social-links"
          >
            <a
              className="fa fa-facebook"
              href="facebook.com"
              target="_blank"
            />
            <a
              className="fa fa-github"
              href="github.com"
              target="_blank"
            />
            <a
              className="fa fa-envelope"
              href="vnovick@gmail.comm"
              target="_blank"
            />
          </div>
        </div>
      </div>
      <div
        className="overlay"
      />
    </div>
    <section
      className="App-subheader"
    >
      <div
        className="App-hero"
      >
        <h3
          className="heading heading-3"
        >
          Login
        </h3>
      </div>
    </section>
  </header>
  <section
    className="App-body"
  >
    <div
      className="raised-2 section"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "center",
              "padding": "2rem",
            }
          }
        >
          <div
            className="input-field form-dirty"
          >
            <input
              label="User Name"
              name="userName"
              onChange={[Function]}
              required={true}
              type="text"
              value="Marik"
            />
            <label>
              User Name
            </label>
          </div>
          <div
            className="input-field form-dirty"
          >
            <input
              label="Password"
              name="password"
              onChange={[Function]}
              required={true}
              type="password"
              value="M"
            />
            <label>
              Password
            </label>
          </div>
          <button
            type="submit"
          >
            Login
          </button>
        </div>
        <div
          className=""
        />
      </form>
    </div>
  </section>
  <footer
    className="App-footer"
  >
    <div
      className="social-links"
    >
      <a
        className="text-link"
        href="facebook.com"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="text-link"
        href="github.com"
        target="_blank"
      >
        Github
      </a>
      <a
        className="text-link"
        href="vnovick@gmail.comm"
        target="_blank"
      >
        Email
      </a>
    </div>
  </footer>
</div>
`;
